src/main/java/org/utplsql/cli/RunCommand.java

```diff
@@ -221,6 +228,48 @@ public String getCommand() {
     }
 
 
+    private void outputMainInformation() {
+
+        StringBlockFormatter formatter = new StringBlockFormatter("utPLCSL cli");
+        formatter.appendLine(CliVersionInfo.getInfo());
+        formatter.appendLine(JavaApiVersionInfo.getInfo());
+        formatter.appendLine("Java-Version: " + System.getProperty("java.version"));
+        formatter.appendLine("ORACLE_HOME: " + EnvironmentVariableUtil.getEnvValue("ORACLE_HOME"));
+        formatter.appendLine("");
+        formatter.appendLine("Thanks for testing!");
+
+        logger.info(formatter.toString());
+        logger.info("");
+    }
+
+    private void initDatabase(DataSource dataSource) throws SQLException {
+        try (Connection conn = dataSource.getConnection()) {
+
+            // Check if orai18n exists if database version is 11g
+            RunCommandChecker.checkOracleI18nExists(conn);
+
+            // First of all do a compatibility check and fail-fast
+            compatibilityProxy = checkFrameworkCompatibility(conn);
+
+            logger.info("Successfully connected to database. UtPLSQL core: " + compatibilityProxy.getDatabaseVersion());
+            logger.info("Oracle-Version: {}", new DefaultDatabaseInformation().getOracleVersion(conn));
```

Comment:
is it possible to provide it as lambda so we don't execute it if not needed?

Reply:
Please clarify, I don't get it

Reply:
something like this: 
```
logger.debug("I am logging that {} happened.", () -> compute());
```
or maybe 
```
logger.debug(() -> "I am logging that "+compute()+" happened.")
```
so db call gather information is only done when logging level is low enough.

Reply:
Neat, didn't know this usage.

Reply:
It's not possible at the moment.
See https://jira.qos.ch/browse/SLF4J-371